subsys/bluetooth/host/conn.c

```diff
@@ -1044,9 +1071,13 @@ void bt_conn_tx_processor(void)
 		destroy_and_callback(conn, buf, cb, ud);
 		bt_conn_disconnect(conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
 
+		bt_conn_unref(conn);
 		return;
 	}
 
+	/* Also need to unref for the happy path */
+	bt_conn_unref(conn);
```

Comment:
Maybe it'd be a bit cleaner to have something like this at the end of the function, assuming it's ok to swap `irq_raise()` and `unref()`:
```c
	bt_tx_irq_raise();
done:
	bt_conn_unref(conn);
}
```

And then `goto done;` in all the branches where you're needing to return from the function.

Reply:
should be ok to swap as this fn always runs from the same thread.

Reply:
done